### PR TITLE
Nezih

### PR DESCRIPTION
**Pull request by MrNezihBJ**
All entities except Demande d'intervention

### CODE REVIEW

File: src/clients/dto/update-client.input.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { CreateClientInput } from './create-client.input';
+import { InputType, Field, Int, PartialType } from '@nestjs/graphql';
+
```
Comment (SkanderLassoued09) [inline src/clients/dto/update-client.input.ts]: Please remove unused files

File: src/clients/entities/client.entity.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { ObjectType, Field, Int } from '@nestjs/graphql';
```
Comment (SkanderLassoued09) [inline src/clients/entities/client.entity.ts]: Please change code : 
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { HydratedDocument } from 'mongoose';

export type CatDocument = HydratedDocument<Cat>; // you can remove this line

@Schema()
export class Cat { // CatDocument
  @Prop()
  name: string;

  @Prop()
  age: number;

  @Prop()
  breed: string;
}

export const CatSchema = SchemaFactory.createForClass(Cat); // CatDocument

File: src/clients/entities/client.entity.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { IsBoolean, IsEmail, IsPhoneNumber, IsString } from 'class-validator';
+import mongoose from 'mongoose';
+
+export type ClientDocument = Client & Document;
+export const ClientSchema = new mongoose.Schema(
+  {
```
Comment (SkanderLassoued09) [inline src/clients/entities/client.entity.ts]: Please change code : 
import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
import { HydratedDocument } from 'mongoose';

export type CatDocument = HydratedDocument<Cat>; // you can remove this line

@Schema()
export class Cat { // CatDocument
  @Prop()
  name: string;

  @Prop()
  age: number;

  @Prop()
  breed: string;
}

export const CatSchema = SchemaFactory.createForClass(Cat); // CatDocument

File: src/companies/companies.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { Resolver, Query, Mutation, Args, Int } from '@nestjs/graphql';
+import { CompaniesService } from './companies.service';
+import { Companie } from './entities/company.entity';
+import { CreateCompanieInput } from './dto/create-company.input';
+
+@Resolver(() => Companie)
```
Comment (SkanderLassoued09) [inline src/companies/companies.resolver.ts]: Please change companie with company

File: src/companies/companies.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+export class CompaniesResolver {
+  constructor(private readonly companiesService: CompaniesService) {}
+
+  @Mutation(() => Companie)
+  async createCompanie(
```
Comment (SkanderLassoued09) [inline src/companies/companies.resolver.ts]: Please change companie with company

File: src/companies/companies.service.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import { Injectable } from '@nestjs/common';
+import { CreateCompanieInput } from './dto/create-company.input';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Companie } from './entities/company.entity';
+
+@Injectable()
+export class CompaniesService {
```
Comment (SkanderLassoued09) [inline src/companies/companies.service.ts]: Please change companie with company

File: src/companies/dto/create-company.input.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { InputType, Field } from '@nestjs/graphql';
+import { service_Achat_Tech_Financier } from '../entities/company.entity';
+import { IsBoolean, IsEmail, IsPhoneNumber, IsString } from 'class-validator';
+
+@InputType()
+export class CreateCompanieInput {
```
Comment (SkanderLassoued09) [inline src/companies/dto/create-company.input.ts]: Please change companie with company

File: src/companies/dto/create-company.input.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  activiteSecondaire: string;
+  @Field()
+  raisonSociale: string;
+  @Field({ nullable: true })
+  Exoneration: string;
+  @Field({ nullable: true })
+  @IsPhoneNumber()
+  fax: string;
+  @Field({ nullable: true })
+  webSiteLink: string;
```
Comment (SkanderLassoued09) [inline src/companies/dto/create-company.input.ts]: Please remove unused comments

File: src/companies/entities/company.entity.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
+import { ObjectType, Field } from '@nestjs/graphql';
+import { IsBoolean, IsEmail, IsPhoneNumber, IsString } from 'class-validator';
+import mongoose, { Document } from 'mongoose';
+
+export type companiesDocument = Companie & Document;
+export const CompanieSchema = new mongoose.Schema(
+  {
```
Comment (SkanderLassoued09) [inline src/companies/entities/company.entity.ts]: Same new declaration as mentioned before

File: src/companies/entities/company.entity.ts [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+      email: String,
+      phone: String,
+    },
+    isDeleted: Boolean,
+  },
+  { _id: false, timestamps: true },
+);
+
+@ObjectType()
+export class service_Achat_Tech_Financier {
```
Comment (SkanderLassoued09) [inline src/companies/entities/company.entity.ts]: Class name should be in Majuscule

File: src/companies/entities/company.entity.ts [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+  @Field({ nullable: true })
+  @IsEmail()
+  email: string;
+  @Field({ nullable: true })
+  @IsPhoneNumber()
+  phone: string;
+}
+
+@ObjectType()
+export class Companie extends Document {
```
Comment (SkanderLassoued09) [inline src/companies/entities/company.entity.ts]: Please change companie with company

File: src/composant/dto/create-composant.input.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
+import { InputType, Int, Field } from '@nestjs/graphql';
+import { IsDate } from 'class-validator';
+
+@InputType()
+export class CreateComposantInput {
+  @Field()
```
Comment (SkanderLassoued09) [inline src/composant/dto/create-composant.input.ts:6]: All fields should be in english or in frensh please

File: src/composant/dto/update-composant.input.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import { CreateComposantInput } from './create-composant.input';
+import { InputType, Field, Int, PartialType } from '@nestjs/graphql';
```
Comment (SkanderLassoued09) [inline src/composant/dto/update-composant.input.ts]: Remove unused comments

File: src/composant/entities/composant.entity.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
+import { ObjectType, Field, Int } from '@nestjs/graphql';
+import { IsDate } from 'class-validator';
+import mongoose from 'mongoose';
+
+export type ComposantDocument = Composant & Document;
```
Comment (SkanderLassoued09) [inline src/composant/entities/composant.entity.ts]: Same declaration as mention before

File: src/composant/entities/composant.entity.ts [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+  prix_vente: number;
+  @Field({ nullable: true })
+  @IsDate()
+  coming_date: Date;
+  @Field({ nullable: true })
+  link: string;
+  @Field({ nullable: true })
+  quantity_stocked: number;
+  @Field()
+  pdf: string;
```
Comment (SkanderLassoued09) [inline src/composant/entities/composant.entity.ts:64]: Specify which PDF you talk about

File: src/composant/entities/composant.entity.ts [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+  @Field()
+  status: string;
```
Comment (SkanderLassoued09) [inline src/composant/entities/composant.entity.ts:66]: Specify which status you talk about

File: src/composant_categorie/composant_categorie.module.ts [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+import { Composant_CategorieService } from './composant_categorie.service';
+import { MongooseModule } from '@nestjs/mongoose';
+import { Composant_CategorieSchema } from './entities/composant_categorie.entity';
+
+@Module({
+  providers: [Composant_CategorieResolver, Composant_CategorieService],
+  imports: [
+    MongooseModule.forFeature([
+      {
+        name: 'Composant_Categorie',
```
Comment (SkanderLassoued09) [inline src/composant_categorie/composant_categorie.module.ts]: Please change categorie by category

File: src/composant_categorie/composant_categorie.resolver.ts [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+import { CreateComposant_CategorieInput } from './dto/create-composant_categorie.input';
+import { Composant_CategorieService } from './composant_categorie.service';
+
+@Resolver(() => Composant_Categorie)
+export class Composant_CategorieResolver {
+  constructor(
+    private readonly composant_CategorieService: Composant_CategorieService,
+  ) {}
+
+  @Mutation(() => Composant_Categorie)
```
Comment (SkanderLassoued09) [inline src/composant_categorie/composant_categorie.resolver.ts]: please choose one approach ComposantCategory or composant_category

File: src/composant_categorie/composant_categorie.resolver.ts [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+    return this.composant_CategorieService.createComposant_Categorie(
+      createComposant_CategorieInput,
+    );
+  }
+
+  @Mutation(() => Boolean)
+  removeComposant_Categorie(@Args('_id') _id: string): Promise<Boolean> {
+    try {
+      return this.composant_CategorieService.removeComposant_Categorie(_id);
+    } catch (error) {
```
Comment (SkanderLassoued09) [inline src/composant_categorie/composant_categorie.resolver.ts]: Remove try and catch

File: src/composant_categorie/entities/composant_categorie.entity.ts [hunk 20]
```diff
@@ -0,0 +1,19 @@
+import { ObjectType, Field, Int } from '@nestjs/graphql';
+import mongoose from 'mongoose';
+
+export type ComposantCategorieDocumet = Composant_Categorie & Document;
+export const Composant_CategorieSchema = new mongoose.Schema(
```
Comment (SkanderLassoued09) [inline src/composant_categorie/entities/composant_categorie.entity.ts]: Same declaration as mentioned before

**Review by SkanderLassoued09 — CHANGES_REQUESTED** [review]
You should follow one approach in writing variables and classes 
You should you comments to define every field and function

File: src/companies/companies.resolver.ts (hunk 5 above)
Comment (MrNezihBJ) [reply src/companies/companies.resolver.ts]: Done

File: src/companies/companies.resolver.ts (hunk 4 above)
Comment (MrNezihBJ) [reply src/companies/companies.resolver.ts]: Done

**Review by SkanderLassoued09 — APPROVED** [review]

**Pull request merged**